src/shared/services/bolt/boltConnection.ts

```diff
@@ -50,44 +49,24 @@ export const hasMultiDbSupport = async (): Promise<boolean> => {
 
 export const validateConnection = (
   driver: Driver | null,
-  res: (driver: Driver) => void,
-  rej: (error?: any) => void
+  resolve: (driver: Driver) => void,
+  reject: (error?: any) => void
 ): void => {
   if (driver === null) {
-    rej()
+    reject()
     return
   }
 
   driver
-    .supportsMultiDb()
-    .then((multiDbSupport: boolean) => {
-      if (!driver || !driver.session) return rej('No connection')
-      const session = driver.session({
-        defaultAccessMode: neo4j.session.READ,
-        database: multiDbSupport ? 'system' : undefined
-      })
-      const txFn = buildTxFunctionByMode(session)
-      txFn &&
-        txFn((tx: { run: (query: string) => void }) =>
-          tx.run('CALL db.indexes()')
-        )
-          .then(() => {
-            session.close()
-            res(driver)
-          })
-          .catch((e: { code: string; message: string }) => {
-            session.close()
-            // Only invalidate the connection if not available
-            // or not authed
-            // or credentials have expired
-            if (!e.code || isBoltConnectionErrorCode(e.code)) {
-              rej(e)
-            } else {
-              res(driver)
-            }
-          })
+    .verifyConnectivity()
+    .then((result: ServerInfo) => {
+      console.log(result)
+      resolve(driver)
+    })
+    .catch(e => {
+      if (!e.code || isBoltConnectionErrorCode(e.code)) reject(e)
```

Comment:
I think it'd be good to keep the comment note about why we only care about isBoltConnectionErrorCOde

Reply:
Also reading through the driver source to see if we still need to do this check

Reply:
Looking at the driver code -> https://github.dev/neo4j/neo4j-javascript-driver/blob/6388d879ceb2b0dba91c2551bfe0868c16694ae8/packages/core/src/driver.ts 
it seems this method can throw without setting it's code to any of the ones we categorize as "boltConnectionErrorCodes". I'm starting to think we should always reject on error, regardless of errorcode